specs/vm/instruction_set.md

```diff
@@ -99,12 +99,14 @@ Some instructions may _panic_, i.e. enter an unrecoverable state. Additionally,
 
 On a non-predicate panic, append a receipt to the list of receipts, modifying `tx.receiptsRoot`:
 
-| name   | type          | description                                                               |
-|--------|---------------|---------------------------------------------------------------------------|
-| `type` | `ReceiptType` | `ReceiptType.Panic`                                                       |
-| `id`   | `byte[32]`    | Contract ID of current context if in an internal context, zero otherwise. |
-| `pc`   | `uint64`      | Value of register `$pc`.                                                  |
-| `is`   | `uint64`      | Value of register `$is`.                                                  |
+| name     | type          | description                                                                    |
+|----------|---------------|--------------------------------------------------------------------------------|
+| `type`   | `ReceiptType` | `ReceiptType.Panic`                                                            |
+| `id`     | `byte[32]`    | Contract ID of current context if in an internal context, zero otherwise.      |
+| `pc`     | `uint64`      | Value of register `$pc`.                                                       |
+| `is`     | `uint64`      | Value of register `$is`.                                                       |
+| `reason` | `uint8`       | The panic reason identifier                                                    |
+| `data`   | `byte[32]`    | ContractId or AssetID associated with the panic if applicable, zero otherwise. |
```

Comment:
Hmmm. So two things:

1. Including a canonical panic reason would make the specs way more verbose. It was actually removed in #256!
2. `data` isn't really needed in the _canonical_ receipt. Especially given that the primary usecase of the requested feature is wallets simulating txs, where the simulated txs don't actually end up on-chain.

I would simply omit any changes to the canonical receipt and only keep the JSON receipt changes.

Reply:
So in essence we can add metadata fields to the panic receipt as an implementation detail, as long as it's not included into the blockchain state or vm/ivg logic?

Reply:
`:yes:`

Reply:
It might be good to distinguish between optional and non-optional fields.

Reply:
Removed from VM rules